### PR TITLE
Next smaller Element Using Stack

### LINKED ISSUE #1746: [Idea]: Finding the Next Smaller Element Using Stack
### Idea Title

Finding the Next Smaller Element Using Stack

### Idea Description

I would like to propose an implementation for finding the next smaller element for each element in an array using a stack.

### Potential Benefits

- [X] Improves user experience
- [X] Enhances learning
- [X] Adds a new feature
- [x] Improves code quality
- [x] Other

### Implementation Suggestions (Optional)

_No response_

### PR DESCRIPTION
**Pull request by riyaa060**
### Description
This pull request adds detailed documentation explaining the algorithm to find the next smaller element for each element in an array using a stack.
Fixes #1746 

### Type of change

- [x] New feature (non-breaking change which adds functionality)
- [x] Documentation update

### Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: docs/Stack/Next_Smaller_Element_Using_Stack
Comment (ajay-dhangar) [inline docs/Stack/Next_Smaller_Element_Using_Stack:1]: save your file with `.md` extension

File: docs/DSA-Problem-Solution/Add two numbers represented as linked lists.md [hunk 1]
```diff
@@ -0,0 +1,237 @@
...
+     \text{sum} = \text{digit1} + \text{digit2} + \text{carry}
+     \]
+   - The new digit to be added to the result is obtained using:
+     \[
+     \text{new digit} = \text{sum} \mod 10
+     \]
+   - The carry for the next position is:
+     \[
+     \text{carry} = \text{sum} // 10
+     \]
```
Comment (ajay-dhangar) [inline docs/DSA-Problem-Solution/Add two numbers represented as linked lists.md:39]: not a good way you need try math formula like `$$ ... $$`

File: docs/DSA-Problem-Solution/Add two numbers represented as linked lists.md [hunk 2]
```diff
@@ -0,0 +1,237 @@
+---
+id: add-two-numbers-as-linked-lists
+title: "Add two numbers represented as linked lists."
+sidebar_label:  Adding Two Numbers Represented as Linked Lists
```
Comment (ajay-dhangar) [inline docs/DSA-Problem-Solution/Add two numbers represented as linked lists.md:4]: You need to delete the write space because we need only one, not two.

**Review by ajay-dhangar — CHANGES_REQUESTED** [review]
update your code based on suggestions

**Comment by riyaa060** [comment]
I have saved file with extension md

**Pull request closed without merging**